zgw/documenten-api/src/main/kotlin/com/ritense/portal/documentenapi/web/rest/DocumentContentResource.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2015-2023 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ritense.portal.documentenapi.web.rest
+
+import com.ritense.portal.documentenapi.client.DocumentenApiClient
+import com.ritense.portal.documentenapi.domain.Document
+import com.ritense.portal.documentenapi.service.DocumentenApiService
+import java.util.UUID
+import kotlinx.coroutines.runBlocking
+import org.springframework.core.io.buffer.DataBuffer
+import org.springframework.http.HttpHeaders
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import org.springframework.http.codec.multipart.FilePart
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestPart
+import org.springframework.web.bind.annotation.RestController
+import reactor.core.publisher.Flux
+
+@RestController
+@RequestMapping(value = ["/api"])
+class DocumentContentResource(
+    val documentenApiClient: DocumentenApiClient,
+    val documentenApiService: DocumentenApiService
+) {
+
+    @GetMapping(value = ["/document/{documentId}/content"])
+    fun downloadStreaming(@PathVariable documentId: UUID): ResponseEntity<Flux<DataBuffer>> {
+        // Request service to get the file's data stream
+        val fileDataStream = documentenApiClient.getDocumentContentStream(documentId)
+
+        val document = runBlocking { documentenApiService.getDocument(documentId) }
+
+        val responseHeaders = HttpHeaders()
+        responseHeaders.set("Content-Disposition", "attachment; filename=\"${document.bestandsnaam}\"")
+
+        return ResponseEntity
```

Comment:
please use `responseHeaders.use { }` so that it closes the stream properly

Reply:
`use {}` method is not avialable for this class.
also, improvements are out of scope, the story was only for splitting the module.